src/wpantund/NCPInstanceBase-Addresses.cpp

```diff
@@ -874,9 +895,12 @@ NCPInstanceBase::on_mesh_prefix_was_removed(Origin origin, const struct in6_addr
 		if (lookup_address_for_prefix(&address, prefix.get_prefix(), prefix.get_length())
 				&& entry.is_slaac() && entry.is_on_mesh() && prefix.get_length() == 64
 		) {
+			if (mOnMeshPrefixes.count(prefix) == 0 || !has_slaac_on_mesh_prefix(origin, prefix))
```

Comment:
Not sure if we should care where the origin is here in the call to `has_slaac_on_mesh_prefix`.
I think if we do have a matching on-mesh prefix with slaac from any origin, we should keep the SLAAC address.

Another point I noticed, if user explicitly added an IPv6 address which matched the prefix (which is actually what weave does), then we should not assume it's slaac address and remove it.

Reply:
Let's fix this up in a separate PR.

Reply:
I am working on PR to supplement this.